File: python/sort/insertion.py
''' Sort items with insertion sort. '''
def sort(data: list) -> None:
    # Iterate through the list `len(data) - 1` times
    for i in range(1, len(data)):
        # Iterate from elemt `i` to 0
        for j in reversed(range(i)):
            # Swap elements if in the wrong order, stop if in order
            if data[j] > data[j + 1]:
                data[j], data[j + 1] = data[j + 1], data[j]
            else:
                break

File: rust/src/sort/insertion.rs
/// Sorts the data using insertion sort
pub fn sort<T: PartialOrd>(data: &mut [T]) {
    // Iterate through the list `data.len() - 1` times
    for i in 1..data.len() {
        // Iterate through the list, using the iteration count as start index
        for j in (i..data.len()).rev() {
            // Swap the elements if in the wrong order
            if data[j - 1] > data[j] {
                data.swap(j - 1, j);
            }
        }
    }
}
